src/lib/reasoners/adt_rel.ml

```diff
@@ -134,7 +136,10 @@ module Domains = struct
         We don't store domains for constructors and selectors. *)
 
     enums: SX.t;
-    (** Set of tracked representatives of enum type. *)
+    (** Set of tracked representatives whose the domain only contains
+        enum constructors, that is constructors without payload.
+
+        We can split on these values after asserting new formulas. *)
```

Comment:
Let's simply mention case splits or the `case_split` function here. The specific condition under which we are performing case splits is an implementation detail of the SAT solver that the theories know nothing about, and "after asserting new formulas" only correspond to the behavior for a specific value of the `--case-split-policy` option.